src/cli/commands/import.js

```diff
@@ -280,7 +280,7 @@ export class Import extends Install {
     if (manifest.name && this.resolver instanceof ImportPackageResolver) {
       this.resolver.rootName = manifest.name;
     }
-    await this.resolver.init(requests, this.flags.flat);
+    await this.resolver.init(requests, this.flags.flat, this.flags.frozenLockfile);
```

Comment:
Damn! I actually did it that way first and then went back to this monstrosity :(